### PR TITLE
Enabled cilium_bpf_map_pressure metric by default

### LINKED ISSUE #22401: Enable cilium_bpf_map_pressure metric by default
Several users have reported that bpf_map_pressure is not exported by default in the metrics, but it would be very useful to be able to debug issues in their cluster.

### PR DESCRIPTION
**Pull request by vishal-chdhry**
This PR enables cilium_bpf_map_pressure metric by default

Signed-off-by: Vishal Choudhary <contactvishaltech@gmail.com>

Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Thanks for contributing!

Fixes: #22401

### CODE REVIEW

**Review by joestringer — CHANGES_REQUESTED** [review]
Looks good thanks! Could you also add a section to `Documentation/operations/upgrade.rst` for `Changed Metrics` and mention that this `cilium_bpf_map_pressure` is now enabled by default?

**Comment by joestringer** [comment]
The docs build failed: https://github.com/cilium/cilium/actions/runs/3609202818/jobs/6082249045

Could you resolve that failure and then squash all the changes into one commit? All of the proposed changes here are related to one particular underlying effort so I think it's reasonable to keep them all together. This would also avoid breaking the docs lint/etc for intermediate commits.

**Review by tommyp1ckles — APPROVED** [review]

File: Documentation/observability/metrics.rst [hunk 1]
```diff
@@ -301,7 +301,7 @@ Name                                       Labels
 ========================================== ===================================================================== ========== ========================================================
 ``bpf_syscall_duration_seconds``           ``operation``, ``outcome``                                            Disabled   Duration of eBPF system call performed
 ``bpf_map_ops_total``                      ``mapName`` (deprecated), ``map_name``, ``operation``, ``outcome``    Enabled    Number of eBPF map operations performed. ``mapName`` is deprecated and will be removed in 1.10. Use ``map_name`` instead.
-``bpf_map_pressure``                       ``map_name``                                                          Disabled   Map pressure defined as fill-up ratio of the map. Policy maps are exceptionally reported only when ratio is over 0.1.
+``bpf_map_pressure``                       ``map_name``                                                          Enabled    Map pressure defined as fill-up ratio of the map. Policy maps are exceptionally reported only when ratio is over 0.1.
```
Comment (tommyp1ckles) [inline Documentation/observability/metrics.rst]: > Policy maps are exceptionally reported only when ratio is over 0.1.

Might be worth improving the wording on this while we're here, it's a bit confusing.

Comment (joestringer) [reply Documentation/observability/metrics.rst]: Here's a suggestion:

``` 
Map pressure defined as a ratio of the map usage compared to its size. Policy map metrics are only reported when the ratio is over 0.1, ie 10% full.
```

File: Documentation/operations/upgrade.rst [hunk 2]
```diff
@@ -363,11 +363,16 @@ Removed Metrics/Labels
...
+~~~~~~~~~~~~~
+
+* ``cilium_bpf_map_pressure`` is now enabled by default.
+
 Helm Options
 ~~~~~~~~~~~~
 
 * The way Linux capabilities are configured has been revamped in this release. 
-  All capabilities of every container in the ``cilium-agent`` DaemonSet is
+  All capabilities of everuiy container in the ``cilium-agent`` DaemonSet is
```
Comment (tommyp1ckles) [inline Documentation/operations/upgrade.rst]: typo

**Review by tommyp1ckles — CHANGES_REQUESTED** [review]

File: Documentation/operations/upgrade.rst (hunk 2 above)
Comment (vishal-chdhry) [reply Documentation/operations/upgrade.rst]: @tommyp1ckles fixed!

**Comment by joestringer** [comment]
Please squash all of the commits together into the first commit. Also, the documentation check is failing, see the link in the "checks" section at the bottom of the page for `Documentation Updates / Validate & Build HTML`.

**Comment by Vishal-Chdhry** [comment]
@joestringer can you help me with squashing the commits part, when i squash it is also somehow including commits that I havent made

**Comment by joestringer** [comment]
@Vishal-Chdhry sure, maybe you can try following a guide like this one? https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec .

**Comment by Vishal-Chdhry** [comment]
Screenshot 2023-01-12 at 23 26 15
@joestringer The commit history of the branch looks like this, there are a lot of commits that arent mine, because of merging of master branch into the branch and when i try to squash it, using the github desktop app it doesnt let me do that

and the guide doesn't really work because the interactive rebase doesn't show my commits, and i cant understand what to squash

**Comment by joestringer** [comment]
@Vishal-Chdhry did you try `git rebase -i origin/master`?

**Comment by joestringer** [comment]
I've marked this draft until the PR is rebased and commits squashed together into a single commit. Once that's done, feel free to click the "Ready for review" button at the bottom of the page or raise attention to this PR again on Slack. Thanks!

**Comment by joestringer** [comment]
Hi @Vishal-Chdhry , any plans to revisit this PR?

**Pull request closed without merging**